doc/instruction_decode_execute.rst

```diff
@@ -0,0 +1,111 @@
+
+The execute block contains the ALU and the multiplier/divider blocks, it does little beyond wiring and instantiating these blocks.
+
+Arithmetic Logic Unit (ALU)
+---------------------------
+Source File: :file:`rtl/ibex_alu.sv`
+
+The Arithmetic Logic Logic (ALU) is a purely combinational block that implements operations required for the Integer Computational Instructions and the comparison operations required for the Control Transfer Instructions in the RV32I RISC-V Specification.
+Other blocks use the ALU for the following tasks:
+
+* Mult/Div uses it to perform addition as part of the multiplication and division algorithms
+* It computes branch targets with a PC + Imm calculation
+* It computes memory addresses for loads and stores with a Reg + Imm calculation
+* The LSU uses it to increment addresses when performing two accesses to handle an unaligned access
+
+.. _mult-div:
+
+Multiplier/Divider Block (MULT/DIV)
+-----------------------------------
+Source Files: :file:`rtl/ibex_multdiv_slow.sv` :file:`rtl/ibex_multdiv_fast.sv`
+
+The Multiplier/Divider (MULT/DIV) is a state machine driven block to perform multiplication and division.
+The fast and slow versions differ in multiplier only, both implement the same form of long division algorithm.
+The ALU block is used by the long division algorithm in both the fast and slow blocks.
+
+Fast Multiplier
+  - Completes multiply in 3-4 cycles using a MAC (multiply accumulate) which is capable of a 17-bit x 17-bit multiplication with a 34-bit accumulator.
+  - A MUL instruction takes 3 cycles, MULH takes 4.
+  - This MAC is internal to the mult/div block (no external ALU use).
+  - Beware it is simply implemented with the ``*`` and ``+`` operators so results heavily depend upon the synthesis tool used.
+  - In some cases it may be desirable to replace this with a specific implementation (such as a hard macro in an FPGA or an explicit gate level implementation).
+
+Slow Multiplier
+  - Completes multiply in 33 cycles using a Baugh-Wooley multiplier (for both MUL and MULH).
+  - The ALU block is used to compute additions.
+
+Divider
+  Both the fast and slow blocks use the same long division algorithm, it takes 37 cycles to compute (though only requires 2 cycles when there is a divide by 0) and proceeds as follows:
+
+    - Cycle 0: Check for divide by 0
+    - Cycle 1: Compute absolute value of operand A (or return result on divide by 0)
+    - Cycle 2: Compute absolute value of operand B
+    - Cycles 4 - 36: Perform long division as described here: https://en.wikipedia.org/wiki/Division_algorithm#Integer_division_(unsigned)_with_remainder.
+
+Control and Status Register Block (CSR)
+---------------------------------------
+Source File: :file:`rtl/ibex_cs_registers.sv`
+
+The CSR contains all of the CSRs (control/status registers).
+Any CSR read/write is handled through this block.
+Performance counters are held in this block and incremented when appropriate (this includes mcycle and minstret).
```

Comment:
I would add "``" before and after mcycle and minstret as we do for all CSRs in the entire doc.